ci-operator/step-registry/hypershift/mce/install/hypershift-mce-install-commands.sh

```diff
@@ -120,32 +119,40 @@ for ((i=1; i<=20; i++)); do
   echo "Waiting for CSV to be ready"
 done
 
-if [ $_apiReady -eq 1 ]; then
-  # Enable Hypershift Preview
-  oc apply -f - <<EOF
+if [ $_apiReady -eq 0 ]; then
+  echo "multiclusterengine subscription could not install in the allotted time."
+  exit 1
+fi
+echo "multiclusterengine installed successfully"
+
+# Enable Hypershift Preview
+oc apply -f - <<EOF
 apiVersion: multicluster.openshift.io/v1
 kind: MultiClusterEngine
 metadata:
   name: multiclusterengine-sample
 spec: {}
 EOF
-  oc annotate mce multiclusterengine-sample imageRepository=quay.io/acm-d
-  echo "multiclusterengine installed successfully"
-  sleep 5
-else
-  echo "multiclusterengine subscription could not install in the allotted time."
-  exit 1
-fi
+sleep 5
 
 oc patch mce multiclusterengine-sample --type=merge -p '{"spec":{"overrides":{"components":[{"name":"hypershift-preview","enabled": true}]}}}'
 
 # It takes some time for this api to become available.
 # So we try multiple times until it succeeds
-# wait for hypershift operator to come online
 _localClusterReady=0
 set +e
 for ((i=1; i<=10; i++)); do
-  oc get managedcluster local-cluster -o 'jsonpath={.status.conditions[?(@.type=="ManagedClusterConditionAvailable")].status}' >> /dev/null
+  oc apply -f - <<EOF
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  labels:
+    local-cluster: "true"
+  name: local-cluster
+spec:
+  hubAcceptsClient: true
+  leaseDurationSeconds: 60
+EOF
   if [ $? -eq 0 ]; then
     _localClusterReady=1
     break
```

Comment:
do we not need to wait for the ManagedClusterConditionAvailable condition anymore? As the logic is now, we're just applying the ManagedCluster and saying it's successful if the `apply` succeeds.

Reply:
@LiangquanLi930 The output of these commands should be `True` if the local-cluster ManagedCluster is available.

```
oc get managedcluster local-cluster -o jsonpath='{.status.conditions[?(@.type=="HubAcceptedManagedCluster")].status}'

oc get managedcluster local-cluster -o jsonpath='{.status.conditions[?(@.type=="ManagedClusterImportSucceeded")].status}'

oc get managedcluster local-cluster -o jsonpath='{.status.conditions[?(@.type=="ManagedClusterConditionAvailable")].status}'

oc get managedcluster local-cluster -o jsonpath='{.status.conditions[?(@.type=="ManagedClusterJoined")].status}'
```